src/nile/core/deploy.py

```diff
@@ -17,18 +26,27 @@ def deploy(
 ):
     """Deploy StarkNet smart contracts."""
     logging.info(f"🚀 Deploying {contract_name}")
+
     base_path = (
         overriding_path if overriding_path else (BUILD_DIRECTORY, ABIS_DIRECTORY)
     )
     register_abi = abi if abi is not None else f"{base_path[1]}/{contract_name}.json"
 
-    output = run_command(
-        operation="deploy",
-        network=network,
-        contract_name=contract_name,
-        overriding_path=overriding_path,
-        inputs=arguments,
-        mainnet_token=mainnet_token,
+    contract = f"{base_path[0]}/{contract_name}.json"
+    command_args = ["--contract", contract]
+
+    if arguments:
+        command_args.append("--inputs")
+        command_args.extend(prepare_params(arguments))
+
+    if mainnet_token:
+        command_args.append("--token")
+        command_args.extend(mainnet_token)
+
+    args = set_args(network)
```

Comment:
can't we abstract these out from the core commands even if `run_command` is gone?

Reply:
The short answer is: probably yes. I didn't think it made sense to do so prior to all the recent conflict fixes, but it might now. I'll look into it